File: src/main/java/me/leodev/trollingfreedomreborn/commands/InvertWalk.java
package me.leodev.trollingfreedomreborn.commands;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

import java.util.ArrayList;

public class InvertWalk implements Listener {

    public static ArrayList<String> Invert1 = new ArrayList<>();

    public static void Invert(final Player p) {
        Invert1.add(p.getName());
    }

    public static void UnInvert(final Player p) {
        if (Invert1.contains(p.getName())) {
            p.setWalkSpeed(0.2f);
            p.setFlySpeed(0.1f);
            Invert1.remove(p.getName());
        }
    }

    @EventHandler(priority = EventPriority.HIGHEST)
    public void onMove(final PlayerMoveEvent move) {
        if (Invert1.contains(move.getPlayer().getName())) {
            move.getPlayer().setFlySpeed(0.5f);
            move.getPlayer().setWalkSpeed(1.0f);
            final Location from = move.getFrom();
            final Location to = move.getTo();
            final double xDiff = from.getX() - to.getX();
            final double zDiff = from.getZ() - to.getZ();
            to.setX(from.getX() + xDiff);
            to.setZ(from.getZ() + zDiff);
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/movement/InvertWalk.kt
package com.leomadrassi.trollingfreedomreborn.trolls.movement

import org.bukkit.Location
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class InvertWalk : Listener {
    companion object {
        val Invert1: MutableList<String> = ArrayList()

        @JvmStatic
        fun Invert(p: Player) {
            Invert1.add(p.name)
        }

        @JvmStatic
        fun UnInvert(p: Player) {
            if (Invert1.contains(p.name)) {
                p.walkSpeed = 0.2f
                p.flySpeed = 0.1f
                Invert1.remove(p.name)
            }
        }
    }

    @EventHandler(priority = EventPriority.HIGHEST)
    fun onMove(move: PlayerMoveEvent) {
        if (Invert1.contains(move.player.name)) {
            move.player.flySpeed = 0.5f
            move.player.walkSpeed = 1.0f
            val from = move.from
            val to = move.to!!
            val xDiff = from.x - to.x
            val zDiff = from.z - to.z
            to.setX(from.x + xDiff)
            to.setZ(from.z + zDiff)
        }
    }
}
